### PR TITLE
Fixes #258 & remove redundant dispose code

### LINKED ISSUE #258: Crash on modern ping packet
The TrueCraft server seems to crash when it is pinged by a modern Minecraft client.

TODO:Screenshot.

### PR DESCRIPTION
**Pull request by FICTURE7**
This should fix that issue and the current networking code is an over complication
in my opinion.

### CODE REVIEW

File: TrueCraft/RemoteClient.cs [hunk 1]
```diff
@@ -326,37 +326,47 @@ private void ProcessNetwork(SocketAsyncEventArgs e)
...
-                    {
-                        Log("Unhandled packet {0}", packet.GetType().Name);
+                }
+
+                var packets = PacketReader.ReadPackets(this, e.Buffer, e.Offset, e.BytesTransferred);
+                try
+                {
+                    foreach (IPacket packet in packets)
+                    {
+                        //LastSuccessfulPacket = packet;
```
Comment (SirCmpwn) [inline TrueCraft/RemoteClient.cs]: Don't comment out code, just remove it.

File: TrueCraft/RemoteClient.cs [hunk 2]
```diff
@@ -326,37 +326,47 @@ private void ProcessNetwork(SocketAsyncEventArgs e)
...
+                        {
+                            Log("Unhandled packet {0}", packet.GetType().Name);
+                        }
                     }
                 }
+                catch (NotSupportedException)
+                {
+                    // Usually thrown when we do not have the requested packet definition/type.
+
+                    // Might want to create its own Exception type for being more specific.
```
Comment (SirCmpwn) [inline TrueCraft/RemoteClient.cs]: Don't leave this kind of commentary in the code

**Comment by SirCmpwn** [comment]
Looks good. Mandatory question: have you ever read any decompiled code from official Minecraft software?

**Comment by FICTURE7** [comment]
Nope.

**Comment by SirCmpwn** [comment]
Cool, thanks!

**Pull request merged**